DBSCE/Overview.md

```diff
@@ -0,0 +1,219 @@
+
+1. **SignIn Succeeds with binding (steps 10-14)**: At this point, all DBSC(E) specific steps are completed. The server returns signed in content with a special header response to the browser: `Sec-Session-Registration` indicated the session is expected to be DBSC compliant. All steps further are as per the original DBSC proposal with additional params introduced for DBSC(E) customization.
+
+### DBSC(E) use cases
+
+This section expands on the [generic design](#high-level-design) to address different enterprise use cases:
+
+#### IDP is RP and Calls Public Local Key Helper
+
+This is the same use case elaborated in the high level design [above](#high-level-design). However, we have separated the IdP/RP components of the server for the enterprise use case in the below diagram.
+
+![IDPSameAsRP-CallsPublicLocalKeyHelper](./IDPSameAsRP-CallsPublicLocalKeyHelper.svg)
+
+Highlights:
+
+- Auth tokens are not mentioned in the DBSC(E) high level design, as they are optional for the consumer use case. However, for the enterprise use case, the IdP can generate the auth tokens and embed the thumbprint of the public key in the token.
+- The token can be delivered to the RP directly through an API instead of a header based response.
+- The tokens also contain the thumpbrint of the public key, and the RP can validate the thumbprint of the public key from the token.
+
+#### IDP Calls Public Local Key Helper
+
+In Enterprise, it is a valid use case to have separate servers as [RP](#relying-party-rp) and [IdP](#identity-provider-idp). We address the use case where these are separate entities and probably from different vendors below.
+
+For easy mapping with the existing DBSC proposal, please note:
+
+- Steps 1-16 specify the key generation process for a public local key helper.
+- Steps 17-29 are [DBSC](https://github.com/wicg/dbsc), added for completeness.
+
+![IDPCallsPublicLocalKeyHelper](./IDPCallsPublicLocalKeyHelper.svg)
+
+#### IDP Calls Private Local Key Helper
+
+A special case is for enterprises that already have `well-known` Local Key Helpers, which are expected to be trusted and enabled by default in a browser. Here, since the browser can trust a given IDP (based on the URL and a policy mapping) and can trust the IDP to invoke the appropriate Local Key Helper (refer [Local key helper on Windows](./KeyGeneration.md#local-key-helper-on-windows)), there are a few optimizations that can be made to the protocol, in skipping the `nonce`(challenge) and reducing the number of round trips between the IDP and the Local Key Helper.
```

Comment:
We should be able to make those optimizations for other platforms too, because our IDP trusts our LocalKeyHelpers. I'd suggest rewording to clarify where the trust is coming from (it's between IDP and specific LocalKeyHelper).

Reply:
Yes. This was meant to be generic, where is the confusion? Is it the sequence diagram that is specific to our headers?

Reply:
It's just the wording of the paragraph. How it reads is that we can do optimizations, because there's a policy for a given IDP, not because IDP trusts LocalKeyHelper. This part:

>>> Here, since the browser can trust a given IDP (based on the URL and a policy mapping) and can trust the IDP to invoke the appropriate Local Key Helper (refer [Local key helper on Windows](./KeyGeneration.md#local-key-helper-on-windows)), there are a few optimizations that can be made to the protocol...

Reply:
Combining mine and Sasha's feedback below, here's some suggested rewording:

A private local key helper is a special use case. It can only be used by a specific IDP that owns its implementation. Because both the IDP and local key helper are built by the same company, they can use a private protocol to communicate with each other. 

In the DBSC(E), private key helper pattern enables following optimizations:

1. Because IDP will trust its private key helper, key helper discovery can be optimized by caching. IDP can specify its preferred ordered list of key helpers in any response and ask browser to cache its preference for a specific amount of time (HelperCacheTime). Such a request doesn’t have to be initiated by the RP that implements DBSC(E), it allows IDP to specify its private key helpers in any communication within the browser context.
2. If IDP has its own proprietary mechanism to ensure that request is coming from a registered device, then it can accept long-lived binding statements without requiring fresh nonce challenge for those. In that case, threat actor will not be able to replay a long-lived binding statement from a different device due to the IDP requirement to provide fresh proof of device. Browser will cache those binding statements for further performance optimization. 
3. RP can specify that it wants to start a DBSC session during the redirect to the IDP to initiate sign-in operation. In this case, if both key generation and sign in finish successfully, when IDP redirects back to RP, browser will intercept that redirect. Browser will then request local key helper to generate new JWT proof and append that JWT proof to the response before redirecting back to the RP. This allows RP to avoid an additional roundtrip to start a DBSC session.

Reply:
Ack. Let me rework the flow diagram and then rephrase the optimizations with more detail.

Reply:
Rephrased a little. Let us review this closely.

Reply:
Done.